src/pages/index.js

```diff
@@ -1,93 +1,242 @@
+);
 const addCardButton = document.querySelector(".profile__add-button");
 const addModal = "#add-modal";
+const editProfileImgForm = document.forms["edit-profile-image-form"];
 const addCardForm = document.forms["add-card-form"];
 const previewModal = "#preview-modal";
+const deleteModal = "#delete-modal";
 
-//add card section to DOM & add initial cards
+const api = new Api({
+  baseUrl: apiConfig.baseUrl,
+  headers: apiConfig.headers,
+});
 
-function createCard(cardData) {
-  const card = new Card(cardData, "#card-template", imageClick);
-  return card.getView();
-}
+// Create user info instance
+const userInfo = new UserInfo(
+  {
+    userName: profileName,
+    userJob: profileDescription,
+  },
+  profileImageEl
+);
+
+//receive user info
+api
+  .getUserInfo()
+  .then((res) => {
+    userInfo.setUserInfo({
+      userName: res.name,
+      userJob: res.about,
+      userImg: res.avatar,
+    });
+  })
+  .catch((err) => {
+    console.error("Failed to receive user info", err);
+  });
+
+//add card section to DOM & add initial cards
 
 const cardSection = new Section(
   {
-    items: initialCards,
-    renderer: (cardData) => {
+    renderer: (cardData, method) => {
+      if (cardData.link) {
+        cardData.cardLink = cardData.link;
+      }
+      if (cardData.name) {
+        cardData.cardName = cardData.name;
+      }
+
       const cardElement = createCard(cardData);
-      cardSection.addItem(cardElement, "append");
+      cardSection.addItem(cardElement, method);
     },
   },
   ".cards__list"
 );
 
-cardSection.renderItems();
+api
+  .getInitialCards()
+  .then((cards) => {
+    cardSection.renderItems(cards, "append");
+  })
+  .catch((err) => {
+    console.error("Failed to receive card data", err);
+  });
+
+function createCard(cardData) {
+  const card = new Card(
+    cardData,
+    "#card-template",
+    imageClick,
+    deleteClick,
+    likeClick
+  );
+  return card.getView();
+}
 
 //Submit handling functions
 
-function handleProfileFormSubmit(input) {
-  userInfo.setUserInfo(input);
-  profileEditPopup.close();
+function handleProfileFormSubmit(input, button) {
+  api
+    .setUserInfo(input)
+    .then((data) => {
+      userInfo.setUserInfo({
+        userName: data.name,
+        userJob: data.about,
+        userImg: data.avatar,
+      });
+      profileEditPopup.close();
+    })
+    .catch((err) => {
+      console.error("Failed to save changes in user info", err);
+    })
+    .finally(() => {
+      button.textContent = "Save";
+    });
 }
 
-function handleAddCardFormSubmit(input) {
-  const cardElement = createCard(input);
-  cardSection.addItem(cardElement, "prepend");
-  addCardPopup.close();
-  addCardForm.reset();
-  formValidators["add-card-form"].disableButton();
+function handleProfileImgFormSubmit(input, button) {
+  api
+    .editProfileImg(input.imageLink)
+    .then((data) => {
+      if (data && data.avatar) {
+        userInfo.setUserInfo({ userImg: data.avatar });
```

Comment:
* When I edit the `avatar` , the profile data disappears from the page https://snipboard.io/Nx3OhM.jpg 

Please, padd `userName, userJob, userImg` here because `setUserInfo` requieres 3 of them

Reply:
fixed